salt/state.py

```diff
@@ -2772,9 +2785,9 @@ def check_requisite(self, low, running, chunks, pre=False):
                         req = {"id": req}
                     req = trim_req(req)
                     found = False
+                    req_key = next(iter(req))
+                    req_val = req[req_key]
                     for chunk in chunks:
-                        req_key = next(iter(req))
-                        req_val = req[req_key]
                         if req_val is None:
                             continue
```

Comment:
This is going to be hit `len(chunks)` times if `req_val is None`. Can we skip the loop completely if this is the case?

Reply:
Good catch.  Thanks!